gnomad/utils/vep.py

```diff
@@ -423,6 +444,36 @@ def filter_vep_to_synonymous_variants(
     )
 
 
+def filter_vep_to_gene_list(
```

Comment:
1st problem was slow to run, 2nd problem was that I don't get same number of variants: 
```
# Only run the test on a few genes

test_genes = [
    "ENSG00000130203",
    "ENSG00000102468",
    "ENSG00000187642",
    "ENSG00000215788",
]

test_intervals = [
    hl.parse_locus_interval("19:45409011-45412650"),  # ENSG00000130203, APOE
    hl.parse_locus_interval("13:47405685-47471169"),  # ENSG00000102468, HTR2A
    hl.parse_locus_interval("1:910579-917497"),  # ENSG00000187642, C1orf170/PERM1
    hl.parse_locus_interval("1:6521211-6526255"),  # ENSG00000215788, TNFRSF25
]

ht = vep_context.ht()
logger.info("There are %d variants in the GRCh37-Gencode v19 vep context_ht", ht.count())

ht1 = hl.filter_intervals(ht, test_intervals)
logger.info("Filteringg vep context_ht to %d variants by filter_intervals...", ht1.count())

ht2 = preprocess_variants_for_tx(ht, filter_to_genes=test_genes)
logger.info("Filteringg vep context_ht to %d variants by filtering to gene list...", ht2.count())
```
INFO (pext_v2_pipeline 3): There are 8575974294 variants in the GRCh37-Gencode v19 vep context_ht
INFO (pext_v2_pipeline 6): Filteringg vep context_ht to 243255 variants by filter_intervals...
wrote table with 673976 rows in 10 partitions to gs://gnomad-tmp-4day/__iruid_128256-hGh5hkJNsfx8l6u1sjtSpi

Reply:
for 1. Yes it will be slower than an interval filter because it's filtering based on the transcript consequences, and doesn't require a separate interval list. To make it faster, the user can first filter their HT to an interval list, and this will further filter the transcripts to the specific gene names 

For 2. When I run that code I get `Filteringg vep context_ht to 30581 variants by filtering to gene list...`. I'm not surprised there are fewer variants when using  `preprocess_variants_for_tx` because it also filters transcripts based on other reasons, it's not only filtering to variants in the specified genes. Using `filter_vep_to_gene_list(ht, genes=test_genes, filter_empty_csq=True)` rather than `preprocess_variants_for_tx` I get `363267` variants.

Since `filter_vep_to_gene_list` is specifically a filter for the VEP consequences of a gene. This can include things like `downstream_gene_variant`. To get the same number you got with the interval filter, you could follow it up with something like (Note: I get 243267 variants, so there might be another csq I am missing in the filter):
```
csq_set = hl.set(['upstream_gene_variant', 'downstream_gene_variant'])

ht2 = ht2.annotate(
    csqs_coding=ht2.vep.transcript_consequences.filter(
        lambda x: hl.any(
            x.consequence_terms.map(
                lambda csq: ~csq_set.contains(csq)
            )
        )
    )
)
ht2_coding = ht2.filter(hl.len(ht2.csqs_coding) > 0)
ht2_coding.count()
```
or directly use `filter_vep_transcript_csqs` with `csqs=['upstream_gene_variant', 'downstream_gene_variant']` and `keep_csqs=False`

Also note that the interval filter only filters the variants to the specified intervals, it doesn't also filter the VEP transcript consequences to those for the specific gene list, it also requires the user specify the intervals which makes the assumption that those intervals are the same as the intervals used by the version of VEP on the HT/MT.

Reply:
Thank you, I should have thought about this, because these intervals might overlap with other genes too. Also thanks for pointing out `['upstream_gene_variant', 'downstream_gene_variant']` that won't be in the gene intervals.

Reply:
Maybe add these 2 things in as notes to the function?